docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -174,10 +174,13 @@ following fields defined:
 
 #### Optional `status` fields
 
-The `status` object **may** define several fields that do not affect functionality if missing:
+The `status` object **may** define several fields:
 
 * `failureReason` - is a string that explains why an error has occurred, if possible.
 * `failureMessage` - is a string that holds the message contained by the error.
+* `externalManagedControlPlane` - is a bool that should be set to true if the Node objects do not
+  exist in the cluster. For example, managed control plane providers for AKS, EKS, GKE, etc, should
```

Comment:
One small note: the AKS implementation doesn't have a separate control plane provider, see https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/19e80de4a7e92082ca0c5eb672e486801da23766/templates/cluster-template-aks.yaml

Would that still work as a field on the AzureManagedControlPlane https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/995da71c822c201cfa72cf99ae1b96652544e352/exp/api/v1alpha3/azuremanagedcontrolplane_types.go#L79 as part of the Azure infra provider?

Reply:
(I think it should since we're using controlPlaneRef the same way)

Reply:
@CecileRobertMichon Yes, as long as you reference the `AzureManagedControlPlane` in the cluster's `controlPlaneRef` field you should be able to use the status in your resource as well. Since the check here in CAPI is using the unstructured API, the underlying type is not important. Whatever object is referenced by `controlPlaneRef` just needs to have the status field `externalManagedControlPlane` if there are no control plane `Machine` resources in the cluster.